### ISSUE TITLE
Weechat should generate appdata.xml metadata

### ISSUE THREAD

**Issue by khaytsus** [weechat/weechat#2051]
### Feature description

Gnome Software and KDE Discovery are two popular tools that users utilize to find software in their distributions.  Regardless of the distribution used, both of these rely on a metadata file which gives information on an application in order to show them in each respective app.  Otherwise, apps such as weechat simply don't show up in Gnome Software or KDE Discovery.   Argument can be made that since it's not a GUI it doesn't need to show up in such applications, however, it may help users find weechat in their distributions if they're only using GUI software install apps to find what's available.  The users may otherwise think weechat is not provided by their distro and ultimately build it from source.

Ideally the software upstream owns this file and maintains it, includes change log information, etc, and the individual package maintainers for distributions can include this file in their packages and ultimately get included in that distribution.

https://www.freedesktop.org/software/appstream/docs/chap-AppStream-About.html#about-whatis describes the overall system and how it works.

https://www.freedesktop.org/software/appstream/docs/chap-CatalogData.html describes the technical aspects of this file.  

Basically if weechat can create and maintain this file, and distributions/maintainers of packages pick it up, then weechat will show up in Gnome-Software or KDE Discovery.

**Comment by flashcode** [comment]
Hi,

WeeChat already provides a desktop file, which is mentioned in your second link, isn't that enough?
Link: https://github.com/weechat/weechat/blob/master/weechat.desktop

If not enough, do you have examples of xml files of other projects?
And isn't the information redundant with the .desktop file?

**Comment by khaytsus** [comment]
Desktop file is not the same thing as the appdata.xml file which is what Gnome Software and KDE Discovery use to generate a directory of applications.  They don't use desktop files for that.

Here is an example of such a file, no particular reason for this example other than it's the one I was given when doing some research into why a user in a Fedora IRC channel was unable to find weechat in KDE Discovery which lead me down the rabbit hole of how it and Gnome Software populate their available software directories.

https://github.com/hexchat/hexchat/blob/master/data/misc/io.github.Hexchat.appdata.xml.in
